src/theme/DocSidebar/index.js

```diff
@@ -0,0 +1,147 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { useState } from 'react';
+import clsx from 'clsx';
+import {
+  useThemeConfig,
+  useAnnouncementBar,
+  MobileSecondaryMenuFiller,
+  ThemeClassNames,
+  useScrollPosition,
+  useWindowSize,
+  DocSidebarItemsExpandedStateProvider,
+} from '@docusaurus/theme-common';
+import Logo from '@theme/Logo';
+import IconArrow from '@theme/IconArrow';
+import { translate } from '@docusaurus/Translate';
+import DocSidebarItems from '@theme/DocSidebarItems';
+import styles from './styles.module.css';
+import Link from '@docusaurus/core/lib/client/exports/Link';
+import PioneerLink from '@site/src/components/PioneerLink';
+
+function useShowAnnouncementBar() {
+  const { isActive } = useAnnouncementBar();
+  const [showAnnouncementBar, setShowAnnouncementBar] = useState(isActive);
+  useScrollPosition(
+    ({ scrollY }) => {
+      if (isActive) {
+        setShowAnnouncementBar(scrollY === 0);
+      }
+    },
+    [isActive]
+  );
+  return isActive && showAnnouncementBar;
+}
+
+function HideableSidebarButton({ onClick }) {
+  return (
+    <button
+      type="button"
+      title={translate({
+        id: 'theme.docs.sidebar.collapseButtonTitle',
+        message: 'Collapse sidebar',
+        description: 'The title attribute for collapse button of doc sidebar',
+      })}
+      aria-label={translate({
+        id: 'theme.docs.sidebar.collapseButtonAriaLabel',
+        message: 'Collapse sidebar',
+        description: 'The title attribute for collapse button of doc sidebar',
+      })}
+      className={clsx(
+        'button button--secondary button--outline',
+        styles.collapseSidebarButton
+      )}
+      onClick={onClick}
+    >
+      <IconArrow className={styles.collapseSidebarButtonIcon} />
+    </button>
+  );
+}
+
+function DocSidebarDesktop({ path, sidebar, onCollapse, isHidden }) {
+  const showAnnouncementBar = useShowAnnouncementBar();
+  const {
+    navbar: { hideOnScroll },
+    hideableSidebar,
+  } = useThemeConfig();
+  return (
+    <div
+      className={clsx(styles.sidebar, {
+        [styles.sidebarWithHideableNavbar]: hideOnScroll,
+        [styles.sidebarHidden]: isHidden,
+      })}
+    >
+      {hideOnScroll && <Logo tabIndex={-1} className={styles.sidebarLogo} />}
+      <nav
+        className={clsx('menu thin-scrollbar', styles.menu, {
+          [styles.menuWithAnnouncementBar]: showAnnouncementBar,
+        })}
+      >
+        <div className={styles.sidebarWrapper}>
+          <div>
+            <ul
+              className={clsx(
+                ThemeClassNames.docs.docSidebarMenu,
+                'menu__list'
+              )}
+            >
+              <DocSidebarItems items={sidebar} activePath={path} level={1} />
+            </ul>
+          </div>
+          <PioneerLink />
+        </div>
+      </nav>
+      {hideableSidebar && <HideableSidebarButton onClick={onCollapse} />}
+    </div>
```

Comment:
Only change in this file is to wrap the `ul` with a `div` so we can use flexbox to push the `PioneerLink` to the bottom of the left nav.